### ISSUE TITLE
add ability to remove protocol from URL 

### ISSUE THREAD

**Issue by eroux** [citation-style-language/schema#395]
**Is your feature request related to a problem? Please describe.**

MLA Handbook 8th edition says in 2.5.2 URLs and DOIs:

> When giving a URL, copy it in full from your Web browser, but omit http:// or https://.

but `modern-language-association.csl` can only copy the URL that is given as argument. 

see
- https://github.com/citation-style-language/styles/issues/5545
- https://discourse.citationstyles.org/t/new-way-to-specify-unprefixed-urls-in-styles/1731

**Describe the solution you'd like**

Add an `npurl` to the list of link renderings:

https://github.com/citation-style-language/documentation/blob/master/specification.rst#appendix-vi-links

that would remove the protocol from a url

*For changes to the CSL RNC schemas, this should include links to citation style guides.* It's better to include links to more than one style guide, and no more than five, for us to assess how widely needed this feature might be.

MLA Handbook 8th edition (published book, not openly available online)

**Comment by bwiernik** [comment]
This is a quirky rule that is unique to MLA as far as I am aware. 

I personally don't really understand why MLA bothers with it and my standard preference is that more clarity rather than less is preferable. For example, many publishers are named "something.com" and including the protocol more clearly distinguishes what is an actual URL where the item is located. 

Given the rarity of this rule, I'm not sure it's worth the additional complexity to ask citation processors to include an option to strip the protocol from a web link. 

What do you think @denismaier @bdarcus @adam3smith ?

**Comment by bdarcus** [comment]
@eroux 

> Add an npurl to the list of link renderings:

So you are not proposing schema changes, but instead for this to be handled in the data? 

If yes, that seems less-than-ideal, given that it's style dependent?

> What do you think @denismaier @bdarcus @adam3smith?

IDK. As you are aware, I'm not fond of many citation rules.

**Comment by adam3smith** [comment]
I'm also skeptical: I have searched around a bit and haven't found a single example of this in a citation style other than MLA

**Comment by eroux** [comment]
well, don't count on me to defend the idea of removing the protocol, I find it quite silly to be honest... but for better or worse, MLA is quite popular in the humanities so that would still be helpful to a number of users

**Comment by bdarcus** [comment]
My worry is where it stops?

Stripping the protocol seems straightforward enough, but are people then
going to ask for other iterations of this?

**Comment by bdarcus** [comment]
BTW, they promote this on twitter; a recent tweet pointing to this:

https://style.mla.org/citing-twitter/

**Comment by bdarcus** [comment]
@andras-simonyi - since you've been dealing with URLs lately, any feedback?

And while I'm at it, any feedback/suggested changes on our related recommendations?

https://github.com/citation-style-language/documentation/blob/master/specification.rst#appendix-vi-links

**Comment by andras-simonyi** [comment]
Well, as for adding something like an attribute to remove the protocol from rendered URL's I don't see big problems from the implementor's point of view. OTOH, I agree that hiding information (which is actually already in a condensed format) is not a good policy, and and can even have security implications, e.g., it can matter whether the protocol is `https` or `http`.

**Comment by denismaier** [comment]
> Well, as for adding something like an attribute to remove the protocol from rendered URL's, I don't see big problems from the implementor's point of view (I wouldn't tinker with the data model).

How would that look like? IIUC, we'd have to introduce a new sibling to `cs:text`, `cs:url` or so, because otherwise the new attribute would be usable in absurd situations. E.g. `<text variable="url" form="no-protocol"/>`  or `<text variable="url" url-form="no-protocol"/>` makes sense, but what about `<text variable="title" form="no-protocol"/>`  or `<text variable="title" url-form="no-protocol"/>`? 

Or could we allow a new value on `@form` or a new attribute `url-form` only when the rendered variable is `url`?

**Comment by georgd** [comment]
> This is a quirky rule that is unique to MLA as far as I am aware.

There surely are some institutional styles, even unrelated to the MLA, with this requirement (I know of one :)). 

And, how could it not be, Austrian legal styles require funny things: either no protocol at all or no protocol unless it is `https`.

**Comment by cormacrelf** [comment]
> And while I'm at it, any feedback/suggested changes on our related recommendations?

Yeah. I think the "if no url rendered in bib entry, add the anchor to the title` fallback thing goes too far. In a nutshell, I don't think anyone wants titles to be blue/underlined but link styling is fine for all the others, and that is quite tricky to resolve at the CSL processor level. More broadly, departing from a convention where every URL is visible in textual form is a bad idea. Think accidental inclusion of incorrect links that someone writing a paper might not even notice have made it in. Think sci-hub.

I can bang out a proper issue for that if you want and discuss there.

**Comment by bdarcus** [comment]
> I can bang out a proper issue for that if you want and discuss there.

I think we should just remove that part. Maybe a PR?

**Comment by bdarcus** [comment]
> How would that look like?

What I was asking about earlier was a likely a global.boolean.

```xml
<citation url-format="strip-protocol">
```

**Comment by denismaier** [comment]
> 
> 
> > How would that look like?
> 
> What I was asking about earlier was a likely a global.boolean.
> 
> ```
> <style url-format="strip-protocol">
> ```

Ok, much better than an attribute on `cs:text`...

**Comment by bwiernik** [comment]
Okay, given that we have found at least a few examples, I think a style attribute makes sense. Could someone open a PR?

**Comment by bdarcus** [comment]
> How should this be handled for rendering of DOI or other identifiers? I suppose that the style would just specify an appropriate prefix in that case.

That's why I mentioned the spec recommendation section.

Was thinking we could rely on this simple attribute plus those recommendations first, and see if and how requirements evolve.
